Add unified.cpu-threshold config (#22167)
pingcap/docs: tikv-configuration-file.md

+ Controls whether to automatically adjust the thread pool size. When it is enabled, the read performance of TiKV is optimized by automatically adjusting the UnifyReadPool thread pool size based on the current CPU usage. The possible range of the thread pool is `[max-thread-count, MAX(4, CPU)]`. The maximum value is the same as the one of [`max-thread-count`](#max-thread-count).
+ Default value: `false`

### `cpu-threshold` <span class="version-mark">New in v8.5.5</span>

> + Specifies the CPU utilization threshold for the unified read pool.

Reviewer: this feature is active only when readpool.unified.auto-adjust is set. https://docs.pingcap.com/tidb/stable/tikv-configuration-file/#auto-adjust-pool-size-new-in-v630

> + Specifies the CPU utilization threshold for the unified read pool.  For example, when this value is set to `0.8`, the thread pool can use up to 80% of the CPU.

Reviewer 2: ```suggestion
+ Specifies the CPU utilization threshold for the unified read pool. For example, if you set this value to `0.8`, the thread pool can use up to 80% of the CPU.
```

    + By default (when it is `0.0`), there is no limit on the CPU usage of UnifyReadPool. The size of the thread pool is determined solely by the busy thread scaling algorithm, which adjusts the size dynamically based on the number of threads handling current tasks.

>     + If it is set to a value greater than `0.0`, TiKV introduces the following CPU usage threshold constraints on top of the existing busy-thread scaling algorithm to control CPU resource usage more strictly:

Reviewer 2: ```suggestion
    + If it is set to a value greater than `0.0`, TiKV applies the following CPU usage threshold constraints in addition to the existing busy-thread scaling algorithm to control CPU resource usage more strictly:
```

> on top of: in addition to something, especially something unpleasant

"in addition to" is preferred because it is more formal and avoids the negative connotation of "on top of"

+ Controls whether to automatically adjust the thread pool size. When it is enabled, the read performance of TiKV is optimized by automatically adjusting the UnifyReadPool thread pool size based on the current CPU usage. The possible range of the thread pool is `[max-thread-count, MAX(4, CPU)]`. The maximum value is the same as the one of [`max-thread-count`](#max-thread-count).
+ Default value: `false`

> ### `cpu-threshold` <span class="version-mark">New in v8.5.5</span>

Reviewer 3: ```suggestion
### `cpu-threshold` <span class="version-mark">New in v8.5.5 and v9.0.0</span>
```

+ Specifies the CPU utilization threshold for the unified read pool.

  - If this parameter is set to `0` (the default), this constraint is disabled, and thread pool scaling is determined solely by the busy thread scaling algorithm.
  - If this parameter is set to a value greater than `0`, CPU threshold checks are imposed on top of thread scaling:
      - A scale-down is forced when CPU usage exceeds the threshold plus a 10% leeway.

>       - A scale-up is prevented if it would cause CPU usage to exceed the threshold minus a 10% leeway.

Reviewer 3: ```suggestion
    + By default (when it is `0.0`), there is no limit on the CPU usage of UnifyReadPool. The size of the thread pool is determined solely by the busy thread scaling algorithm, which adjusts the size dynamically based on the number of threads handling current tasks.
    + If it is set to a value greater than `0.0`, TiKV introduces the following CPU usage threshold constraints on top of the existing busy-thread scaling algorithm to control CPU resource usage more strictly:
        + **Forced scale-down**: when the CPU usage of the unified read pool exceeds the configured value plus a 10% buffer, TiKV forcibly reduces the size of the pool.
        + **Scale-up prevention**: when expanding the unified read pool would cause CPU usage to exceed the configured threshold minus a 10% buffer, TiKV prevents the unified read pool from further expanding.
```
